### ISSUE TITLE
sexp(::Matrix) has copy-paste bug?

### ISSUE THREAD

**Issue by robertfeldt** [JuliaStats/RCall.jl#17]
I can't convert a Julia Matrix{Float64} into sexp:

```julia
feldt:~/.julia/v0.3$ julia
               _
   _       _ _(_)_     |  A fresh approach to technical computing
  (_)     | (_) (_)    |  Documentation: http://docs.julialang.org
   _ _   _| |_  __ _   |  Type "help()" for help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 0.3.5 (2015-01-08 22:33 UTC)
 _/ |\__'_|_|_|\__'_|  |  Official http://julialang.org/ release
|__/                   |  x86_64-apple-darwin13.4.0

julia> using RCall

julia> m = [0.0 1.0; 2.0 3.0]
2x2 Array{Float64,2}:
 0.0  1.0
 2.0  3.0

julia> rm = sexp(m)
ERROR: l not defined
 in sexp at /Users/feldt/.julia/v0.3/RCall/src/sexp.jl:129
```

Seems to me that l should instead be q (or p?) on line 129.

**Comment by robertfeldt** [comment]
Actually I tried with either p or q instead of l but both give BoundsError() so we need you here Douglas :)

```julia
julia> rm = sexp(m)
ERROR: BoundsError()
 in copy! at array.jl:52
 in sexp at /Users/feldt/.julia/v0.3/RCall/src/sexp.jl:129
```

**Comment by robertfeldt** [comment]
Ok, I think I get it now. It should be p*q:

```julia
julia> m = [0.0 1.0; 2.0 3.0]
2x2 Array{Float64,2}:
 0.0  1.0
 2.0  3.0

julia> R.str(reval(lang2(sexp(:mean), sexp(m))))
 num 1.5
SEXP{0}(Ptr{Void} @0x00007fecbee28f78)
```
I will submit a pull request.

**Comment by robertfeldt** [comment]
Note that I have not fixed the sexp implementation for the multi-dim julia arrays. Should be a similar fix I guess but since I'm not sure I leave that to you.

**Comment by dmbates** [comment]
closed by 711ca438e1b61bc796f4b96032adc75d9b5a22ff
